modules/Sources/Dependencies/AddressBookClient/AddressBookLiveKey.swift

```diff
@@ -34,304 +39,208 @@ extension AddressBookClient: DependencyKey {
                 guard latestKnownContacts == nil else {
-                    if let contacts = latestKnownContacts {
-                        return contacts
-                    } else {
-                        return .empty
-                    }
+                    return latestKnownContacts ?? .empty
                 }
 
                 // contacts haven't been loaded from the locale storage yet, do it
                 do {
                     guard let documentsDirectory = FileManager.default.urls(for: .documentDirectory, in: .userDomainMask).first else {
                         throw AddressBookClientError.documentsFolder
                     }
-                    let fileURL = documentsDirectory.appendingPathComponent(Constants.component)
-                    let encryptedContacts = try Data(contentsOf: fileURL)
 
-                    let decryptedContacts = try AddressBookClient.decryptData(encryptedContacts)
-                    latestKnownContacts = decryptedContacts
+                    // Try to find and get the data from the encrypted file with the latest encryption version
+                    let encryptedFileURL = documentsDirectory.appendingPathComponent(try AddressBookClient.filenameForEncryptedFile())
 
-                    return decryptedContacts
+                    if let contactsData = try? Data(contentsOf: encryptedFileURL) {
+                        // file exists, try to find the unencrypted file in case the delete failed
+                        let unencryptedFileURL = documentsDirectory.appendingPathComponent(Constants.unencryptedFilename)
+                        if FileManager.default.fileExists(atPath: unencryptedFileURL.path) {
+                            try? FileManager.default.removeItem(at: unencryptedFileURL)
+                        }
+                        
+                        let contacts = try AddressBookClient.contactsFrom(encryptedData: contactsData)
+
+                        latestKnownContacts = contacts
+                        return contacts
+                    } else {
+                        // Fallback to the unencrypted file check and resolution
+                        let unencryptedFileURL = documentsDirectory.appendingPathComponent(Constants.unencryptedFilename)
+                        
+                        if let contactsData = try? Data(contentsOf: unencryptedFileURL) {
+                            // file exists, ensure data are parsed, re-saved and file deteled
+                            let contacts = try AddressBookClient.contactsFrom(unencryptedData: contactsData)
+
+                            // try to encrypt and store the data
+                            do {
+                                try AddressBookClient.storeContacts(contacts, remoteStorage: remoteStorage, remoteStore: false)
+                            } catch {
+                                // the store of the new file failed, skip the file remove
```

Comment:
This is correct. I updated the design doc to explicitly say that the deletion here should only be done if the store succeeds.